ROX:26971: Add CRS to ACS docs (#89252)
openshift/openshift-docs: modules/crs-apply-secured-cluster.adoc

:cloud-ocp-apply:
endif::[]

ifeval::["{context}" == "init-bundle-cloud-other-apply"]
:cloud-other-apply:
endif::[]

:_mod-docs-content-type: PROCEDURE
[id="crs-apply-secured-cluster_{context}"]
= Applying the cluster registration secret (CRS) on the secured cluster

//Do not show for ACSCS
ifndef::cloud[]
Before you configure a secured cluster, you must apply the CRS to the cluster. After you have applied the CRS, the services on the secured cluster can communicate securely with Central.
endif::cloud[]

//Show for ACSCS
ifdef::cloud[]
Before you configure a secured cluster, you must apply the CRS to the secured cluster. After you have applied the CRS, the services on the secured cluster can communicate securely with {product-title-managed-short}.
endif::cloud[]

[NOTE]
====
If you are installing by using Helm charts, do not perform this step. Complete the installation by using Helm; See "Installing
{product-title-short} on secured clusters by using Helm charts" in the additional resources section.
====

.Prerequisites
* You must have generated a CRS.

.Procedure
//Show for OpenShift
ifdef::openshift[]
To create resources, perform only one of the following steps:

> * Create resources using the {ocp} web console: In the {ocp} web console, go to the `stackrox` project or the project where you want to install the secured cluster services . In the top menu, click *+* to open the *Import YAML* page. You can drag the CRS file or copy and paste its contents into the editor, and then click *Create*. When the command is complete, the display shows that the secret named `cluster-registration-secret` was created.

Reviewer: ```suggestion
* Create resources using the {ocp} web console: In the {ocp} web console, go to the `stackrox` project or the project where you want to install the secured cluster services. In the top menu, click *+* to open the *Import YAML* page. You can drag the CRS file or copy and paste its contents into the editor, and then click *Create*. When the command is complete, the display shows that the secret named `cluster-registration-secret` was created.
```

// * cloud_service/installing_cloud_ocp/init-bundle-cloud-ocp-apply.adoc
// * installing/installing_ocp/init-bundle-ocp.adoc
// * installing/installing_other/init-bundle-other.adoc

//Conditionals to show ocp commands for OpenShift, kube commands for Kubernetes, Central language for non-cloud
ifeval::["{context}" == "init-bundle-cloud-ocp"]
:openshift:
endif::[]

ifeval::["{context}" == "init-bundle-ocp"]
:openshift:
endif::[]

ifeval::["{context}" == "init-bundle-other"]
:kube:
endif::[]

ifeval::["{context}" == "init-bundle-cloud-other-apply"]
:kube:
endif::[]

ifeval::["{context}" == "init-bundle-cloud-other-apply"]
:cloud:
endif::[]

ifeval::["{context}" == "init-bundle-cloud-ocp-apply"]
:openshift:
endif::[]

ifeval::["{context}" == "init-bundle-cloud-ocp-apply"]
:cloud:
endif::[]

ifeval::["{context}" == "init-bundle-cloud-ocp-apply"]
:cloud-ocp-apply:
endif::[]

ifeval::["{context}" == "init-bundle-cloud-other-apply"]
:cloud-other-apply:
endif::[]

:_mod-docs-content-type: PROCEDURE
[id="crs-apply-secured-cluster_{context}"]
= Applying the cluster registration secret (CRS) on the secured cluster

//Do not show for ACSCS
ifndef::cloud[]

> Before you configure a secured cluster, you must apply the CRS by using it to create the required resources on the cluster. Applying the CRS allows the services on the secured cluster to communicate with Central.

Reviewer 2: I thinks is overly complex.

```suggestion
Before you configure a secured cluster, you must apply the CRS to the cluster. This enables the services on the secured cluster to communicate securely with Central.
```

Author: The problem with beginning the sentence with "this" is that it can be unclear what "this" refers to (the act of applying the CRS to the cluster). I will rewrite, but we generally try to avoid that.

Author: Actually, the reason I have it set up that way is that having it in one sentence (Before you configure a secured cluster, you must apply the CRS to the cluster to enable services on the secured cluster.) is too long, and beginning with "this" is an unclear antecedent. I would prefer to leave this as is, unless it's technically inaccurate.

> Before you configure a secured cluster, you must apply the CRS by using it to create the required resources on the cluster. After you apply the CRS, the services on the secured cluster can communicate securely with Central.

Author: Thanks - I think this language was reused from the init bundles, so I will change. Your explanation makes sense!

endif::cloud[]

//Show for ACSCS
ifdef::cloud[]

> Before you configure a secured cluster, you must apply the CRS by using it to create the required resources on the secured cluster. Applying the CRS allows the services on the secured cluster to communicate with {product-title-managed-short}.

Reviewer 2: Same here.

endif::cloud[]

[NOTE]
====
If you are installing by using Helm charts, do not perform this step. Complete the installation by using Helm; See "Installing
{product-title-short} on secured clusters by using Helm charts" in the additional resources section.
====

.Prerequisites
* You must have generated a CRS.

> * You must have created the `stackrox` project, or namespace, on the cluster where secured cluster services will be installed. Using `stackrox` for the project is not required, but ensures that vulnerabilities for {product-title-short} processes are not reported when scanning your clusters.

Reviewer 2: Thinking about [this ticket](https://issues.redhat.com/browse/ROX-10921)...

Given that it seems that there are still open questions about non-standard namespaces (e.g. default policies), maybe we should not advertise this too much?

Author: This was existing text from the init bundle copied over. I do plan to address the whole issue in that other Jira ticket in the rest of the docs so for now I'll just remove this here since it's a new module.

.Procedure
//Show for OpenShift
ifdef::openshift[]
To create resources, perform only one of the following steps:

> * Create resources using the {ocp} web console: In the {ocp} web console, make sure that you are in the `stackrox` namespace. In the top menu, click *+* to open the *Import YAML* page. You can drag the CRS file or copy and paste its contents into the editor, and then click *Create*. When the command is complete, the display shows that the `tls-cert-collector`, `tls-cert-sensor`, and tls-cert-admission-control` resources were created.

Reviewer 2: This is not correct for CRS:
> When the command is complete, the display shows that the `tls-cert-collector`, `tls-cert-sensor`, and tls-cert-admission-control` resources were created.

Instead, there is just one single resource being created, it is if kind `Secret` and is named `cluster-registration-secret`.

* Create resources using the {osp} CLI: Using the {osp} CLI, run the following command to create the resources:
+
[source,terminal]
----

> $ oc create -f <crs_name>.yaml \// <1>

Reviewer 2: what you specify with `-f` is just the file name of the CRS, not the "CRS name".

  -n <stackrox> <2>
----
<1> Specify the file name of the CRS.

> <2> Specify the name of the project where Central services are installed.

Reviewer 2: Not central services, we are on the secured cluster here (Secured cluster services).
